server/src/main/java/org/apache/kafka/server/AssignmentsManager.java

```diff
@@ -231,13 +238,12 @@ public void run() throws Exception {
             } else {
                 failedAttempts = 0;
                 AssignReplicasToDirsResponseData data = ((AssignReplicasToDirsResponse) response.responseBody()).data();
+
                 Set<AssignmentEvent> failed = filterFailures(data, inflight);
+                Set<AssignmentEvent> completed = Utils.diff(HashSet::new, inflight.values().stream().collect(Collectors.toSet()), failed);
+                completed.forEach(assignmentEvent -> assignmentEvent.callback.run());
+
                 log.warn("Re-queueing assignments: {}", failed);
-                if (!failed.isEmpty()) {
-                    for (AssignmentEvent event : failed) {
-                        pending.put(event.partition, event);
-                    }
-                }
```

Comment:
I think we need to requeue these, did this not break the tests in AssignmentsManagerTest?

Reply:
Nope didn't fail.  I think I removed it by mistake during refactoring.